// @tinkoff/invest-js/build/services/InstrumentsService.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.InstrumentsService = void 0;
const InstrumentStatus_1 = require("../generated/tinkoff/public/invest/api/contract/v1/InstrumentStatus");
const InstrumentIdType_1 = require("../generated/tinkoff/public/invest/api/contract/v1/InstrumentIdType");
const constants_1 = require("../constants");
const load_1 = require("../load");
const contract = (0, load_1.load)(constants_1.PROTO_PATH + 'instruments.proto');
class InstrumentsService extends contract.InstrumentsService {
    constructor() {
        super(...arguments);
        /**
         * Метод получения расписания торгов торговых площадок
         */
        this.tradingSchedules = (argument, ...restArgs) => {
            const nowMs = Date.now();
            const nowSeconds = Math.floor(nowMs / 1000);
            const nanos = (nowMs % 1000) * 1e6;
            const defaultArgument = {
                from: { seconds: nowSeconds, nanos },
                to: { seconds: nowSeconds + 6 * 24 * 3600, nanos },
            };
            return super.tradingSchedules(Object.assign(defaultArgument, argument), 
            // @ts-ignore
            ...restArgs);
        };
    }
    /**
     * Метод получения списка акций для базовонго списка инструментов.
     */
    baseShares(callback) {
        return this.shares({ instrumentStatus: InstrumentStatus_1.InstrumentStatus.INSTRUMENT_STATUS_BASE }, callback);
    }
    /**
     * Метод получения списка акций для всех инструментов.
     */
    allShares(callback) {
        return this.shares({ instrumentStatus: InstrumentStatus_1.InstrumentStatus.INSTRUMENT_STATUS_ALL }, callback);
    }
    /**
     * Метод получения основной информации об инструменте c идентификатором FIGI.
     */
    getInstrumentByFIGI(argument, callback) {
        return this.getInstrumentBy(Object.assign({ idType: InstrumentIdType_1.InstrumentIdType.INSTRUMENT_ID_TYPE_FIGI }, argument), callback);
    }
    /**
     * Метод получения основной информации об инструменте c идентификатором Ticker.
     */
    getInstrumentByTicker(argument, callback) {
        return this.getInstrumentBy(Object.assign({ idType: InstrumentIdType_1.InstrumentIdType.INSTRUMENT_ID_TYPE_TICKER }, argument), callback);
    }
}
exports.InstrumentsService = InstrumentsService;
