// @tinkoff/invest-js/build/client2.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.TTechApiClient = void 0;
const nice_grpc_1 = require("nice-grpc");
const instruments_1 = require("./compiled_contracts/instruments");
const orders_1 = require("./compiled_contracts/orders");
const operations_1 = require("./compiled_contracts/operations");
const marketdata_1 = require("./compiled_contracts/marketdata");
const sandbox_1 = require("./compiled_contracts/sandbox");
const stoporders_1 = require("./compiled_contracts/stoporders");
const users_1 = require("./compiled_contracts/users");
class TTechApiClient {
    constructor(config) {
        var token = config.token;
        var url = config.url || 'https://invest-public-api.tinkoff.ru';
        const providedMetadata = config.metadata || {};
        var clientFactory = (0, nice_grpc_1.createClientFactory)()
            .use((call, options) => call.next(call.request, Object.assign(Object.assign({}, options), { metadata: (0, nice_grpc_1.Metadata)(options.metadata).set('Authorization', `Bearer ${token}`) })));
        const channel = (0, nice_grpc_1.createChannel)(url, nice_grpc_1.ChannelCredentials.createSsl());
        this.instruments = clientFactory.create(instruments_1.InstrumentsServiceDefinition, channel);
        this.ordersStream = clientFactory.create(orders_1.OrdersStreamServiceDefinition, channel);
        this.orders = clientFactory.create(orders_1.OrdersServiceDefinition, channel);
        this.operations = clientFactory.create(operations_1.OperationsServiceDefinition, channel);
        this.operationsStream = clientFactory.create(operations_1.OperationsStreamServiceDefinition, channel);
        this.marketdata = clientFactory.create(marketdata_1.MarketDataServiceDefinition, channel);
        this.marketdataStream = clientFactory.create(marketdata_1.MarketDataStreamServiceDefinition, channel);
        this.stopOrders = clientFactory.create(stoporders_1.StopOrdersServiceDefinition, channel);
        this.users = clientFactory.create(users_1.UsersServiceDefinition, channel);
        this.sandbox = clientFactory.create(sandbox_1.SandboxServiceDefinition, channel);
    }
}
exports.TTechApiClient = TTechApiClient;
