// @tinkoff/invest-js/build/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
var _a, _b;
Object.defineProperty(exports, "__esModule", { value: true });
exports.TTechApiClient = exports.OpenAPIClient = exports.MarketDataStreamService = exports.MarketDataService = exports.OrdersStreamService = exports.OrdersService = exports.OperationsService = exports.StopOrdersService = exports.UsersService = exports.SandboxService = void 0;
const grpc_js_1 = require("@grpc/grpc-js");
const constants_1 = require("./constants");
const load_1 = require("./load");
const client2_1 = require("./client2");
Object.defineProperty(exports, "TTechApiClient", { enumerable: true, get: function () { return client2_1.TTechApiClient; } });
const InstrumentsService_1 = require("./services/InstrumentsService");
exports.SandboxService = (0, load_1.load)(constants_1.PROTO_PATH + 'sandbox.proto').SandboxService;
exports.UsersService = (0, load_1.load)(constants_1.PROTO_PATH + 'users.proto').UsersService;
exports.StopOrdersService = (0, load_1.load)(constants_1.PROTO_PATH + 'stoporders.proto').StopOrdersService;
exports.OperationsService = (0, load_1.load)(constants_1.PROTO_PATH + 'operations.proto').OperationsService;
_a = (0, load_1.load)(constants_1.PROTO_PATH + 'orders.proto'), exports.OrdersService = _a.OrdersService, exports.OrdersStreamService = _a.OrdersStreamService;
_b = (0, load_1.load)(constants_1.PROTO_PATH + 'marketdata.proto'), exports.MarketDataService = _b.MarketDataService, exports.MarketDataStreamService = _b.MarketDataStreamService;
class OpenAPIClient {
    constructor(options) {
        this.token = options.token;
        this.url = options.url || 'invest-public-api.tinkoff.ru:443';
        const providedMetadata = options.metadata || {};
        const metadata = new grpc_js_1.Metadata();
        metadata.add('Authorization', 'Bearer ' + this.token);
        for (const providedMetadataKey in providedMetadata) {
            metadata.add(providedMetadataKey, providedMetadata[providedMetadataKey]);
        }
        const ssl_creds = grpc_js_1.credentials.combineChannelCredentials(grpc_js_1.credentials.createSsl(), grpc_js_1.credentials.createFromMetadataGenerator((_, callback) => callback(null, metadata)));
        this.instruments = new InstrumentsService_1.InstrumentsService(this.url, ssl_creds);
        this.ordersStream = new exports.OrdersStreamService(this.url, ssl_creds);
        this.marketDataStream = new exports.MarketDataStreamService(this.url, ssl_creds);
        this.marketData = new exports.MarketDataService(this.url, ssl_creds);
        this.usersService = new exports.UsersService(this.url, ssl_creds);
        this.orders = new exports.OrdersService(this.url, ssl_creds);
        this.operations = new exports.OperationsService(this.url, ssl_creds);
        this.stopOrders = new exports.StopOrdersService(this.url, ssl_creds);
        this.sandbox = new exports.SandboxService(this.url, ssl_creds);
    }
}
exports.OpenAPIClient = OpenAPIClient;
__exportStar(require("./compiled_contracts/instruments"), exports);
__exportStar(require("./compiled_contracts/marketdata"), exports);
__exportStar(require("./compiled_contracts/orders"), exports);
__exportStar(require("./compiled_contracts/operations"), exports);
__exportStar(require("./compiled_contracts/stoporders"), exports);
__exportStar(require("./compiled_contracts/users"), exports);
__exportStar(require("./compiled_contracts/common"), exports);
__exportStar(require("./compiled_contracts/sandbox"), exports);
