// @tinkoff/invest-js/build/compiled_contracts/sandbox.js
"use strict";
// Code generated by protoc-gen-ts_proto. DO NOT EDIT.
// versions:
//   protoc-gen-ts_proto  v2.6.1
//   protoc               v3.19.1
// source: sandbox.proto
Object.defineProperty(exports, "__esModule", { value: true });
exports.SandboxServiceDefinition = exports.SandboxPayInResponse = exports.SandboxPayInRequest = exports.CloseSandboxAccountResponse = exports.CloseSandboxAccountRequest = exports.OpenSandboxAccountResponse = exports.OpenSandboxAccountRequest = void 0;
/* eslint-disable */
const wire_1 = require("@bufbuild/protobuf/wire");
const common_1 = require("./common");
const operations_1 = require("./operations");
const orders_1 = require("./orders");
const users_1 = require("./users");
function createBaseOpenSandboxAccountRequest() {
    return { name: undefined };
}
exports.OpenSandboxAccountRequest = {
    encode(message, writer = new wire_1.BinaryWriter()) {
        if (message.name !== undefined) {
            writer.uint32(10).string(message.name);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof wire_1.BinaryReader ? input : new wire_1.BinaryReader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseOpenSandboxAccountRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1: {
                    if (tag !== 10) {
                        break;
                    }
                    message.name = reader.string();
                    continue;
                }
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return { name: isSet(object.name) ? globalThis.String(object.name) : undefined };
    },
    toJSON(message) {
        const obj = {};
        if (message.name !== undefined) {
            obj.name = message.name;
        }
        return obj;
    },
    create(base) {
        return exports.OpenSandboxAccountRequest.fromPartial(base !== null && base !== void 0 ? base : {});
    },
    fromPartial(object) {
        var _a;
        const message = createBaseOpenSandboxAccountRequest();
        message.name = (_a = object.name) !== null && _a !== void 0 ? _a : undefined;
        return message;
    },
};
function createBaseOpenSandboxAccountResponse() {
    return { accountId: "" };
}
exports.OpenSandboxAccountResponse = {
    encode(message, writer = new wire_1.BinaryWriter()) {
        if (message.accountId !== "") {
            writer.uint32(10).string(message.accountId);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof wire_1.BinaryReader ? input : new wire_1.BinaryReader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseOpenSandboxAccountResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1: {
                    if (tag !== 10) {
                        break;
                    }
                    message.accountId = reader.string();
                    continue;
                }
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return { accountId: isSet(object.accountId) ? globalThis.String(object.accountId) : "" };
    },
    toJSON(message) {
        const obj = {};
        if (message.accountId !== "") {
            obj.accountId = message.accountId;
        }
        return obj;
    },
    create(base) {
        return exports.OpenSandboxAccountResponse.fromPartial(base !== null && base !== void 0 ? base : {});
    },
    fromPartial(object) {
        var _a;
        const message = createBaseOpenSandboxAccountResponse();
        message.accountId = (_a = object.accountId) !== null && _a !== void 0 ? _a : "";
        return message;
    },
};
function createBaseCloseSandboxAccountRequest() {
    return { accountId: "" };
}
exports.CloseSandboxAccountRequest = {
    encode(message, writer = new wire_1.BinaryWriter()) {
        if (message.accountId !== "") {
            writer.uint32(10).string(message.accountId);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof wire_1.BinaryReader ? input : new wire_1.BinaryReader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseCloseSandboxAccountRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1: {
                    if (tag !== 10) {
                        break;
                    }
                    message.accountId = reader.string();
                    continue;
                }
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return { accountId: isSet(object.accountId) ? globalThis.String(object.accountId) : "" };
    },
    toJSON(message) {
        const obj = {};
        if (message.accountId !== "") {
            obj.accountId = message.accountId;
        }
        return obj;
    },
    create(base) {
        return exports.CloseSandboxAccountRequest.fromPartial(base !== null && base !== void 0 ? base : {});
    },
    fromPartial(object) {
        var _a;
        const message = createBaseCloseSandboxAccountRequest();
        message.accountId = (_a = object.accountId) !== null && _a !== void 0 ? _a : "";
        return message;
    },
};
function createBaseCloseSandboxAccountResponse() {
    return {};
}
exports.CloseSandboxAccountResponse = {
    encode(_, writer = new wire_1.BinaryWriter()) {
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof wire_1.BinaryReader ? input : new wire_1.BinaryReader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseCloseSandboxAccountResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(_) {
        return {};
    },
    toJSON(_) {
        const obj = {};
        return obj;
    },
    create(base) {
        return exports.CloseSandboxAccountResponse.fromPartial(base !== null && base !== void 0 ? base : {});
    },
    fromPartial(_) {
        const message = createBaseCloseSandboxAccountResponse();
        return message;
    },
};
function createBaseSandboxPayInRequest() {
    return { accountId: "", amount: undefined };
}
exports.SandboxPayInRequest = {
    encode(message, writer = new wire_1.BinaryWriter()) {
        if (message.accountId !== "") {
            writer.uint32(10).string(message.accountId);
        }
        if (message.amount !== undefined) {
            common_1.MoneyValue.encode(message.amount, writer.uint32(18).fork()).join();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof wire_1.BinaryReader ? input : new wire_1.BinaryReader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseSandboxPayInRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1: {
                    if (tag !== 10) {
                        break;
                    }
                    message.accountId = reader.string();
                    continue;
                }
                case 2: {
                    if (tag !== 18) {
                        break;
                    }
                    message.amount = common_1.MoneyValue.decode(reader, reader.uint32());
                    continue;
                }
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            accountId: isSet(object.accountId) ? globalThis.String(object.accountId) : "",
            amount: isSet(object.amount) ? common_1.MoneyValue.fromJSON(object.amount) : undefined,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.accountId !== "") {
            obj.accountId = message.accountId;
        }
        if (message.amount !== undefined) {
            obj.amount = common_1.MoneyValue.toJSON(message.amount);
        }
        return obj;
    },
    create(base) {
        return exports.SandboxPayInRequest.fromPartial(base !== null && base !== void 0 ? base : {});
    },
    fromPartial(object) {
        var _a;
        const message = createBaseSandboxPayInRequest();
        message.accountId = (_a = object.accountId) !== null && _a !== void 0 ? _a : "";
        message.amount = (object.amount !== undefined && object.amount !== null)
            ? common_1.MoneyValue.fromPartial(object.amount)
            : undefined;
        return message;
    },
};
function createBaseSandboxPayInResponse() {
    return { balance: undefined };
}
exports.SandboxPayInResponse = {
    encode(message, writer = new wire_1.BinaryWriter()) {
        if (message.balance !== undefined) {
            common_1.MoneyValue.encode(message.balance, writer.uint32(10).fork()).join();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof wire_1.BinaryReader ? input : new wire_1.BinaryReader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseSandboxPayInResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1: {
                    if (tag !== 10) {
                        break;
                    }
                    message.balance = common_1.MoneyValue.decode(reader, reader.uint32());
                    continue;
                }
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return { balance: isSet(object.balance) ? common_1.MoneyValue.fromJSON(object.balance) : undefined };
    },
    toJSON(message) {
        const obj = {};
        if (message.balance !== undefined) {
            obj.balance = common_1.MoneyValue.toJSON(message.balance);
        }
        return obj;
    },
    create(base) {
        return exports.SandboxPayInResponse.fromPartial(base !== null && base !== void 0 ? base : {});
    },
    fromPartial(object) {
        const message = createBaseSandboxPayInResponse();
        message.balance = (object.balance !== undefined && object.balance !== null)
            ? common_1.MoneyValue.fromPartial(object.balance)
            : undefined;
        return message;
    },
};
exports.SandboxServiceDefinition = {
    name: "SandboxService",
    fullName: "tinkoff.public.invest.api.contract.v1.SandboxService",
    methods: {
        /** Зарегистрировать счёт. */
        openSandboxAccount: {
            name: "OpenSandboxAccount",
            requestType: exports.OpenSandboxAccountRequest,
            requestStream: false,
            responseType: exports.OpenSandboxAccountResponse,
            responseStream: false,
            options: {},
        },
        /** Получить счета. */
        getSandboxAccounts: {
            name: "GetSandboxAccounts",
            requestType: users_1.GetAccountsRequest,
            requestStream: false,
            responseType: users_1.GetAccountsResponse,
            responseStream: false,
            options: {},
        },
        /** Закрыть счёт. */
        closeSandboxAccount: {
            name: "CloseSandboxAccount",
            requestType: exports.CloseSandboxAccountRequest,
            requestStream: false,
            responseType: exports.CloseSandboxAccountResponse,
            responseStream: false,
            options: {},
        },
        /** Выставить торговое поручение. */
        postSandboxOrder: {
            name: "PostSandboxOrder",
            requestType: orders_1.PostOrderRequest,
            requestStream: false,
            responseType: orders_1.PostOrderResponse,
            responseStream: false,
            options: {},
        },
        /** Изменить выставленную заявку. */
        replaceSandboxOrder: {
            name: "ReplaceSandboxOrder",
            requestType: orders_1.ReplaceOrderRequest,
            requestStream: false,
            responseType: orders_1.PostOrderResponse,
            responseStream: false,
            options: {},
        },
        /** Получить список активных заявок по счёту. */
        getSandboxOrders: {
            name: "GetSandboxOrders",
            requestType: orders_1.GetOrdersRequest,
            requestStream: false,
            responseType: orders_1.GetOrdersResponse,
            responseStream: false,
            options: {},
        },
        /** Отменить торговое поручение. */
        cancelSandboxOrder: {
            name: "CancelSandboxOrder",
            requestType: orders_1.CancelOrderRequest,
            requestStream: false,
            responseType: orders_1.CancelOrderResponse,
            responseStream: false,
            options: {},
        },
        /** Поулчить статус заявки в песочнице. Заявки хранятся в таблице 7 дней. */
        getSandboxOrderState: {
            name: "GetSandboxOrderState",
            requestType: orders_1.GetOrderStateRequest,
            requestStream: false,
            responseType: orders_1.OrderState,
            responseStream: false,
            options: {},
        },
        /** Получить позиции по виртуальному счёту. */
        getSandboxPositions: {
            name: "GetSandboxPositions",
            requestType: operations_1.PositionsRequest,
            requestStream: false,
            responseType: operations_1.PositionsResponse,
            responseStream: false,
            options: {},
        },
        /** Получить операции по номеру счёта. */
        getSandboxOperations: {
            name: "GetSandboxOperations",
            requestType: operations_1.OperationsRequest,
            requestStream: false,
            responseType: operations_1.OperationsResponse,
            responseStream: false,
            options: {},
        },
        /** Получить операции по номеру счёта с пагинацией. */
        getSandboxOperationsByCursor: {
            name: "GetSandboxOperationsByCursor",
            requestType: operations_1.GetOperationsByCursorRequest,
            requestStream: false,
            responseType: operations_1.GetOperationsByCursorResponse,
            responseStream: false,
            options: {},
        },
        /** Получить портфель. */
        getSandboxPortfolio: {
            name: "GetSandboxPortfolio",
            requestType: operations_1.PortfolioRequest,
            requestStream: false,
            responseType: operations_1.PortfolioResponse,
            responseStream: false,
            options: {},
        },
        /** Пополнить счёт. */
        sandboxPayIn: {
            name: "SandboxPayIn",
            requestType: exports.SandboxPayInRequest,
            requestStream: false,
            responseType: exports.SandboxPayInResponse,
            responseStream: false,
            options: {},
        },
        /** Получить доступный остаток для вывода средств. */
        getSandboxWithdrawLimits: {
            name: "GetSandboxWithdrawLimits",
            requestType: operations_1.WithdrawLimitsRequest,
            requestStream: false,
            responseType: operations_1.WithdrawLimitsResponse,
            responseStream: false,
            options: {},
        },
        /** Расчёт количества доступных для покупки/продажи лотов в песочнице. */
        getSandboxMaxLots: {
            name: "GetSandboxMaxLots",
            requestType: orders_1.GetMaxLotsRequest,
            requestStream: false,
            responseType: orders_1.GetMaxLotsResponse,
            responseStream: false,
            options: {},
        },
    },
};
function isSet(value) {
    return value !== null && value !== undefined;
}
