// @tinkoff/invest-js/build/compiled_contracts/users.js
"use strict";
// Code generated by protoc-gen-ts_proto. DO NOT EDIT.
// versions:
//   protoc-gen-ts_proto  v2.6.1
//   protoc               v3.19.1
// source: users.proto
Object.defineProperty(exports, "__esModule", { value: true });
exports.UsersServiceDefinition = exports.GetInfoResponse = exports.GetInfoRequest = exports.StreamLimit = exports.UnaryLimit = exports.GetUserTariffResponse = exports.GetUserTariffRequest = exports.GetMarginAttributesResponse = exports.GetMarginAttributesRequest = exports.Account = exports.GetAccountsResponse = exports.GetAccountsRequest = exports.AccessLevel = exports.AccountStatus = exports.AccountType = void 0;
exports.accountTypeFromJSON = accountTypeFromJSON;
exports.accountTypeToJSON = accountTypeToJSON;
exports.accountStatusFromJSON = accountStatusFromJSON;
exports.accountStatusToJSON = accountStatusToJSON;
exports.accessLevelFromJSON = accessLevelFromJSON;
exports.accessLevelToJSON = accessLevelToJSON;
/* eslint-disable */
const wire_1 = require("@bufbuild/protobuf/wire");
const common_1 = require("./common");
const timestamp_1 = require("./google/protobuf/timestamp");
/** Тип счёта. */
var AccountType;
(function (AccountType) {
    /** ACCOUNT_TYPE_UNSPECIFIED - Тип аккаунта не определён. */
    AccountType[AccountType["ACCOUNT_TYPE_UNSPECIFIED"] = 0] = "ACCOUNT_TYPE_UNSPECIFIED";
    /** ACCOUNT_TYPE_TINKOFF - Брокерский счёт Т-Инвестиций. */
    AccountType[AccountType["ACCOUNT_TYPE_TINKOFF"] = 1] = "ACCOUNT_TYPE_TINKOFF";
    /** ACCOUNT_TYPE_TINKOFF_IIS - ИИС. */
    AccountType[AccountType["ACCOUNT_TYPE_TINKOFF_IIS"] = 2] = "ACCOUNT_TYPE_TINKOFF_IIS";
    /** ACCOUNT_TYPE_INVEST_BOX - Инвесткопилка. */
    AccountType[AccountType["ACCOUNT_TYPE_INVEST_BOX"] = 3] = "ACCOUNT_TYPE_INVEST_BOX";
    /** ACCOUNT_TYPE_INVEST_FUND - Фонд денежного рынка. */
    AccountType[AccountType["ACCOUNT_TYPE_INVEST_FUND"] = 4] = "ACCOUNT_TYPE_INVEST_FUND";
    AccountType[AccountType["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(AccountType || (exports.AccountType = AccountType = {}));
function accountTypeFromJSON(object) {
    switch (object) {
        case 0:
        case "ACCOUNT_TYPE_UNSPECIFIED":
            return AccountType.ACCOUNT_TYPE_UNSPECIFIED;
        case 1:
        case "ACCOUNT_TYPE_TINKOFF":
            return AccountType.ACCOUNT_TYPE_TINKOFF;
        case 2:
        case "ACCOUNT_TYPE_TINKOFF_IIS":
            return AccountType.ACCOUNT_TYPE_TINKOFF_IIS;
        case 3:
        case "ACCOUNT_TYPE_INVEST_BOX":
            return AccountType.ACCOUNT_TYPE_INVEST_BOX;
        case 4:
        case "ACCOUNT_TYPE_INVEST_FUND":
            return AccountType.ACCOUNT_TYPE_INVEST_FUND;
        case -1:
        case "UNRECOGNIZED":
        default:
            return AccountType.UNRECOGNIZED;
    }
}
function accountTypeToJSON(object) {
    switch (object) {
        case AccountType.ACCOUNT_TYPE_UNSPECIFIED:
            return "ACCOUNT_TYPE_UNSPECIFIED";
        case AccountType.ACCOUNT_TYPE_TINKOFF:
            return "ACCOUNT_TYPE_TINKOFF";
        case AccountType.ACCOUNT_TYPE_TINKOFF_IIS:
            return "ACCOUNT_TYPE_TINKOFF_IIS";
        case AccountType.ACCOUNT_TYPE_INVEST_BOX:
            return "ACCOUNT_TYPE_INVEST_BOX";
        case AccountType.ACCOUNT_TYPE_INVEST_FUND:
            return "ACCOUNT_TYPE_INVEST_FUND";
        case AccountType.UNRECOGNIZED:
        default:
            return "UNRECOGNIZED";
    }
}
/** Статус счёта. */
var AccountStatus;
(function (AccountStatus) {
    /** ACCOUNT_STATUS_UNSPECIFIED - Статус счёта не определён. */
    AccountStatus[AccountStatus["ACCOUNT_STATUS_UNSPECIFIED"] = 0] = "ACCOUNT_STATUS_UNSPECIFIED";
    /** ACCOUNT_STATUS_NEW - Новый, в процессе открытия. */
    AccountStatus[AccountStatus["ACCOUNT_STATUS_NEW"] = 1] = "ACCOUNT_STATUS_NEW";
    /** ACCOUNT_STATUS_OPEN - Открытый и активный счёт. */
    AccountStatus[AccountStatus["ACCOUNT_STATUS_OPEN"] = 2] = "ACCOUNT_STATUS_OPEN";
    /** ACCOUNT_STATUS_CLOSED - Закрытый счёт. */
    AccountStatus[AccountStatus["ACCOUNT_STATUS_CLOSED"] = 3] = "ACCOUNT_STATUS_CLOSED";
    /** ACCOUNT_STATUS_ALL - Все счета. */
    AccountStatus[AccountStatus["ACCOUNT_STATUS_ALL"] = 4] = "ACCOUNT_STATUS_ALL";
    AccountStatus[AccountStatus["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(AccountStatus || (exports.AccountStatus = AccountStatus = {}));
function accountStatusFromJSON(object) {
    switch (object) {
        case 0:
        case "ACCOUNT_STATUS_UNSPECIFIED":
            return AccountStatus.ACCOUNT_STATUS_UNSPECIFIED;
        case 1:
        case "ACCOUNT_STATUS_NEW":
            return AccountStatus.ACCOUNT_STATUS_NEW;
        case 2:
        case "ACCOUNT_STATUS_OPEN":
            return AccountStatus.ACCOUNT_STATUS_OPEN;
        case 3:
        case "ACCOUNT_STATUS_CLOSED":
            return AccountStatus.ACCOUNT_STATUS_CLOSED;
        case 4:
        case "ACCOUNT_STATUS_ALL":
            return AccountStatus.ACCOUNT_STATUS_ALL;
        case -1:
        case "UNRECOGNIZED":
        default:
            return AccountStatus.UNRECOGNIZED;
    }
}
function accountStatusToJSON(object) {
    switch (object) {
        case AccountStatus.ACCOUNT_STATUS_UNSPECIFIED:
            return "ACCOUNT_STATUS_UNSPECIFIED";
        case AccountStatus.ACCOUNT_STATUS_NEW:
            return "ACCOUNT_STATUS_NEW";
        case AccountStatus.ACCOUNT_STATUS_OPEN:
            return "ACCOUNT_STATUS_OPEN";
        case AccountStatus.ACCOUNT_STATUS_CLOSED:
            return "ACCOUNT_STATUS_CLOSED";
        case AccountStatus.ACCOUNT_STATUS_ALL:
            return "ACCOUNT_STATUS_ALL";
        case AccountStatus.UNRECOGNIZED:
        default:
            return "UNRECOGNIZED";
    }
}
/** Уровень доступа к счёту. */
var AccessLevel;
(function (AccessLevel) {
    /** ACCOUNT_ACCESS_LEVEL_UNSPECIFIED - Уровень доступа не определён. */
    AccessLevel[AccessLevel["ACCOUNT_ACCESS_LEVEL_UNSPECIFIED"] = 0] = "ACCOUNT_ACCESS_LEVEL_UNSPECIFIED";
    /** ACCOUNT_ACCESS_LEVEL_FULL_ACCESS - Полный доступ к счёту. */
    AccessLevel[AccessLevel["ACCOUNT_ACCESS_LEVEL_FULL_ACCESS"] = 1] = "ACCOUNT_ACCESS_LEVEL_FULL_ACCESS";
    /** ACCOUNT_ACCESS_LEVEL_READ_ONLY - Доступ с уровнем прав «только чтение». */
    AccessLevel[AccessLevel["ACCOUNT_ACCESS_LEVEL_READ_ONLY"] = 2] = "ACCOUNT_ACCESS_LEVEL_READ_ONLY";
    /** ACCOUNT_ACCESS_LEVEL_NO_ACCESS - Доступа нет. */
    AccessLevel[AccessLevel["ACCOUNT_ACCESS_LEVEL_NO_ACCESS"] = 3] = "ACCOUNT_ACCESS_LEVEL_NO_ACCESS";
    AccessLevel[AccessLevel["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(AccessLevel || (exports.AccessLevel = AccessLevel = {}));
function accessLevelFromJSON(object) {
    switch (object) {
        case 0:
        case "ACCOUNT_ACCESS_LEVEL_UNSPECIFIED":
            return AccessLevel.ACCOUNT_ACCESS_LEVEL_UNSPECIFIED;
        case 1:
        case "ACCOUNT_ACCESS_LEVEL_FULL_ACCESS":
            return AccessLevel.ACCOUNT_ACCESS_LEVEL_FULL_ACCESS;
        case 2:
        case "ACCOUNT_ACCESS_LEVEL_READ_ONLY":
            return AccessLevel.ACCOUNT_ACCESS_LEVEL_READ_ONLY;
        case 3:
        case "ACCOUNT_ACCESS_LEVEL_NO_ACCESS":
            return AccessLevel.ACCOUNT_ACCESS_LEVEL_NO_ACCESS;
        case -1:
        case "UNRECOGNIZED":
        default:
            return AccessLevel.UNRECOGNIZED;
    }
}
function accessLevelToJSON(object) {
    switch (object) {
        case AccessLevel.ACCOUNT_ACCESS_LEVEL_UNSPECIFIED:
            return "ACCOUNT_ACCESS_LEVEL_UNSPECIFIED";
        case AccessLevel.ACCOUNT_ACCESS_LEVEL_FULL_ACCESS:
            return "ACCOUNT_ACCESS_LEVEL_FULL_ACCESS";
        case AccessLevel.ACCOUNT_ACCESS_LEVEL_READ_ONLY:
            return "ACCOUNT_ACCESS_LEVEL_READ_ONLY";
        case AccessLevel.ACCOUNT_ACCESS_LEVEL_NO_ACCESS:
            return "ACCOUNT_ACCESS_LEVEL_NO_ACCESS";
        case AccessLevel.UNRECOGNIZED:
        default:
            return "UNRECOGNIZED";
    }
}
function createBaseGetAccountsRequest() {
    return { status: undefined };
}
exports.GetAccountsRequest = {
    encode(message, writer = new wire_1.BinaryWriter()) {
        if (message.status !== undefined) {
            writer.uint32(8).int32(message.status);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof wire_1.BinaryReader ? input : new wire_1.BinaryReader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseGetAccountsRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1: {
                    if (tag !== 8) {
                        break;
                    }
                    message.status = reader.int32();
                    continue;
                }
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return { status: isSet(object.status) ? accountStatusFromJSON(object.status) : undefined };
    },
    toJSON(message) {
        const obj = {};
        if (message.status !== undefined) {
            obj.status = accountStatusToJSON(message.status);
        }
        return obj;
    },
    create(base) {
        return exports.GetAccountsRequest.fromPartial(base !== null && base !== void 0 ? base : {});
    },
    fromPartial(object) {
        var _a;
        const message = createBaseGetAccountsRequest();
        message.status = (_a = object.status) !== null && _a !== void 0 ? _a : undefined;
        return message;
    },
};
function createBaseGetAccountsResponse() {
    return { accounts: [] };
}
exports.GetAccountsResponse = {
    encode(message, writer = new wire_1.BinaryWriter()) {
        for (const v of message.accounts) {
            exports.Account.encode(v, writer.uint32(10).fork()).join();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof wire_1.BinaryReader ? input : new wire_1.BinaryReader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseGetAccountsResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1: {
                    if (tag !== 10) {
                        break;
                    }
                    message.accounts.push(exports.Account.decode(reader, reader.uint32()));
                    continue;
                }
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            accounts: globalThis.Array.isArray(object === null || object === void 0 ? void 0 : object.accounts) ? object.accounts.map((e) => exports.Account.fromJSON(e)) : [],
        };
    },
    toJSON(message) {
        var _a;
        const obj = {};
        if ((_a = message.accounts) === null || _a === void 0 ? void 0 : _a.length) {
            obj.accounts = message.accounts.map((e) => exports.Account.toJSON(e));
        }
        return obj;
    },
    create(base) {
        return exports.GetAccountsResponse.fromPartial(base !== null && base !== void 0 ? base : {});
    },
    fromPartial(object) {
        var _a;
        const message = createBaseGetAccountsResponse();
        message.accounts = ((_a = object.accounts) === null || _a === void 0 ? void 0 : _a.map((e) => exports.Account.fromPartial(e))) || [];
        return message;
    },
};
function createBaseAccount() {
    return { id: "", type: 0, name: "", status: 0, openedDate: undefined, closedDate: undefined, accessLevel: 0 };
}
exports.Account = {
    encode(message, writer = new wire_1.BinaryWriter()) {
        if (message.id !== "") {
            writer.uint32(10).string(message.id);
        }
        if (message.type !== 0) {
            writer.uint32(16).int32(message.type);
        }
        if (message.name !== "") {
            writer.uint32(26).string(message.name);
        }
        if (message.status !== 0) {
            writer.uint32(32).int32(message.status);
        }
        if (message.openedDate !== undefined) {
            timestamp_1.Timestamp.encode(toTimestamp(message.openedDate), writer.uint32(42).fork()).join();
        }
        if (message.closedDate !== undefined) {
            timestamp_1.Timestamp.encode(toTimestamp(message.closedDate), writer.uint32(50).fork()).join();
        }
        if (message.accessLevel !== 0) {
            writer.uint32(56).int32(message.accessLevel);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof wire_1.BinaryReader ? input : new wire_1.BinaryReader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseAccount();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1: {
                    if (tag !== 10) {
                        break;
                    }
                    message.id = reader.string();
                    continue;
                }
                case 2: {
                    if (tag !== 16) {
                        break;
                    }
                    message.type = reader.int32();
                    continue;
                }
                case 3: {
                    if (tag !== 26) {
                        break;
                    }
                    message.name = reader.string();
                    continue;
                }
                case 4: {
                    if (tag !== 32) {
                        break;
                    }
                    message.status = reader.int32();
                    continue;
                }
                case 5: {
                    if (tag !== 42) {
                        break;
                    }
                    message.openedDate = fromTimestamp(timestamp_1.Timestamp.decode(reader, reader.uint32()));
                    continue;
                }
                case 6: {
                    if (tag !== 50) {
                        break;
                    }
                    message.closedDate = fromTimestamp(timestamp_1.Timestamp.decode(reader, reader.uint32()));
                    continue;
                }
                case 7: {
                    if (tag !== 56) {
                        break;
                    }
                    message.accessLevel = reader.int32();
                    continue;
                }
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            id: isSet(object.id) ? globalThis.String(object.id) : "",
            type: isSet(object.type) ? accountTypeFromJSON(object.type) : 0,
            name: isSet(object.name) ? globalThis.String(object.name) : "",
            status: isSet(object.status) ? accountStatusFromJSON(object.status) : 0,
            openedDate: isSet(object.openedDate) ? fromJsonTimestamp(object.openedDate) : undefined,
            closedDate: isSet(object.closedDate) ? fromJsonTimestamp(object.closedDate) : undefined,
            accessLevel: isSet(object.accessLevel) ? accessLevelFromJSON(object.accessLevel) : 0,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.id !== "") {
            obj.id = message.id;
        }
        if (message.type !== 0) {
            obj.type = accountTypeToJSON(message.type);
        }
        if (message.name !== "") {
            obj.name = message.name;
        }
        if (message.status !== 0) {
            obj.status = accountStatusToJSON(message.status);
        }
        if (message.openedDate !== undefined) {
            obj.openedDate = message.openedDate.toISOString();
        }
        if (message.closedDate !== undefined) {
            obj.closedDate = message.closedDate.toISOString();
        }
        if (message.accessLevel !== 0) {
            obj.accessLevel = accessLevelToJSON(message.accessLevel);
        }
        return obj;
    },
    create(base) {
        return exports.Account.fromPartial(base !== null && base !== void 0 ? base : {});
    },
    fromPartial(object) {
        var _a, _b, _c, _d, _e, _f, _g;
        const message = createBaseAccount();
        message.id = (_a = object.id) !== null && _a !== void 0 ? _a : "";
        message.type = (_b = object.type) !== null && _b !== void 0 ? _b : 0;
        message.name = (_c = object.name) !== null && _c !== void 0 ? _c : "";
        message.status = (_d = object.status) !== null && _d !== void 0 ? _d : 0;
        message.openedDate = (_e = object.openedDate) !== null && _e !== void 0 ? _e : undefined;
        message.closedDate = (_f = object.closedDate) !== null && _f !== void 0 ? _f : undefined;
        message.accessLevel = (_g = object.accessLevel) !== null && _g !== void 0 ? _g : 0;
        return message;
    },
};
function createBaseGetMarginAttributesRequest() {
    return { accountId: "" };
}
exports.GetMarginAttributesRequest = {
    encode(message, writer = new wire_1.BinaryWriter()) {
        if (message.accountId !== "") {
            writer.uint32(10).string(message.accountId);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof wire_1.BinaryReader ? input : new wire_1.BinaryReader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseGetMarginAttributesRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1: {
                    if (tag !== 10) {
                        break;
                    }
                    message.accountId = reader.string();
                    continue;
                }
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return { accountId: isSet(object.accountId) ? globalThis.String(object.accountId) : "" };
    },
    toJSON(message) {
        const obj = {};
        if (message.accountId !== "") {
            obj.accountId = message.accountId;
        }
        return obj;
    },
    create(base) {
        return exports.GetMarginAttributesRequest.fromPartial(base !== null && base !== void 0 ? base : {});
    },
    fromPartial(object) {
        var _a;
        const message = createBaseGetMarginAttributesRequest();
        message.accountId = (_a = object.accountId) !== null && _a !== void 0 ? _a : "";
        return message;
    },
};
function createBaseGetMarginAttributesResponse() {
    return {
        liquidPortfolio: undefined,
        startingMargin: undefined,
        minimalMargin: undefined,
        fundsSufficiencyLevel: undefined,
        amountOfMissingFunds: undefined,
        correctedMargin: undefined,
    };
}
exports.GetMarginAttributesResponse = {
    encode(message, writer = new wire_1.BinaryWriter()) {
        if (message.liquidPortfolio !== undefined) {
            common_1.MoneyValue.encode(message.liquidPortfolio, writer.uint32(10).fork()).join();
        }
        if (message.startingMargin !== undefined) {
            common_1.MoneyValue.encode(message.startingMargin, writer.uint32(18).fork()).join();
        }
        if (message.minimalMargin !== undefined) {
            common_1.MoneyValue.encode(message.minimalMargin, writer.uint32(26).fork()).join();
        }
        if (message.fundsSufficiencyLevel !== undefined) {
            common_1.Quotation.encode(message.fundsSufficiencyLevel, writer.uint32(34).fork()).join();
        }
        if (message.amountOfMissingFunds !== undefined) {
            common_1.MoneyValue.encode(message.amountOfMissingFunds, writer.uint32(42).fork()).join();
        }
        if (message.correctedMargin !== undefined) {
            common_1.MoneyValue.encode(message.correctedMargin, writer.uint32(50).fork()).join();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof wire_1.BinaryReader ? input : new wire_1.BinaryReader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseGetMarginAttributesResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1: {
                    if (tag !== 10) {
                        break;
                    }
                    message.liquidPortfolio = common_1.MoneyValue.decode(reader, reader.uint32());
                    continue;
                }
                case 2: {
                    if (tag !== 18) {
                        break;
                    }
                    message.startingMargin = common_1.MoneyValue.decode(reader, reader.uint32());
                    continue;
                }
                case 3: {
                    if (tag !== 26) {
                        break;
                    }
                    message.minimalMargin = common_1.MoneyValue.decode(reader, reader.uint32());
                    continue;
                }
                case 4: {
                    if (tag !== 34) {
                        break;
                    }
                    message.fundsSufficiencyLevel = common_1.Quotation.decode(reader, reader.uint32());
                    continue;
                }
                case 5: {
                    if (tag !== 42) {
                        break;
                    }
                    message.amountOfMissingFunds = common_1.MoneyValue.decode(reader, reader.uint32());
                    continue;
                }
                case 6: {
                    if (tag !== 50) {
                        break;
                    }
                    message.correctedMargin = common_1.MoneyValue.decode(reader, reader.uint32());
                    continue;
                }
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            liquidPortfolio: isSet(object.liquidPortfolio) ? common_1.MoneyValue.fromJSON(object.liquidPortfolio) : undefined,
            startingMargin: isSet(object.startingMargin) ? common_1.MoneyValue.fromJSON(object.startingMargin) : undefined,
            minimalMargin: isSet(object.minimalMargin) ? common_1.MoneyValue.fromJSON(object.minimalMargin) : undefined,
            fundsSufficiencyLevel: isSet(object.fundsSufficiencyLevel)
                ? common_1.Quotation.fromJSON(object.fundsSufficiencyLevel)
                : undefined,
            amountOfMissingFunds: isSet(object.amountOfMissingFunds)
                ? common_1.MoneyValue.fromJSON(object.amountOfMissingFunds)
                : undefined,
            correctedMargin: isSet(object.correctedMargin) ? common_1.MoneyValue.fromJSON(object.correctedMargin) : undefined,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.liquidPortfolio !== undefined) {
            obj.liquidPortfolio = common_1.MoneyValue.toJSON(message.liquidPortfolio);
        }
        if (message.startingMargin !== undefined) {
            obj.startingMargin = common_1.MoneyValue.toJSON(message.startingMargin);
        }
        if (message.minimalMargin !== undefined) {
            obj.minimalMargin = common_1.MoneyValue.toJSON(message.minimalMargin);
        }
        if (message.fundsSufficiencyLevel !== undefined) {
            obj.fundsSufficiencyLevel = common_1.Quotation.toJSON(message.fundsSufficiencyLevel);
        }
        if (message.amountOfMissingFunds !== undefined) {
            obj.amountOfMissingFunds = common_1.MoneyValue.toJSON(message.amountOfMissingFunds);
        }
        if (message.correctedMargin !== undefined) {
            obj.correctedMargin = common_1.MoneyValue.toJSON(message.correctedMargin);
        }
        return obj;
    },
    create(base) {
        return exports.GetMarginAttributesResponse.fromPartial(base !== null && base !== void 0 ? base : {});
    },
    fromPartial(object) {
        const message = createBaseGetMarginAttributesResponse();
        message.liquidPortfolio = (object.liquidPortfolio !== undefined && object.liquidPortfolio !== null)
            ? common_1.MoneyValue.fromPartial(object.liquidPortfolio)
            : undefined;
        message.startingMargin = (object.startingMargin !== undefined && object.startingMargin !== null)
            ? common_1.MoneyValue.fromPartial(object.startingMargin)
            : undefined;
        message.minimalMargin = (object.minimalMargin !== undefined && object.minimalMargin !== null)
            ? common_1.MoneyValue.fromPartial(object.minimalMargin)
            : undefined;
        message.fundsSufficiencyLevel =
            (object.fundsSufficiencyLevel !== undefined && object.fundsSufficiencyLevel !== null)
                ? common_1.Quotation.fromPartial(object.fundsSufficiencyLevel)
                : undefined;
        message.amountOfMissingFunds = (object.amountOfMissingFunds !== undefined && object.amountOfMissingFunds !== null)
            ? common_1.MoneyValue.fromPartial(object.amountOfMissingFunds)
            : undefined;
        message.correctedMargin = (object.correctedMargin !== undefined && object.correctedMargin !== null)
            ? common_1.MoneyValue.fromPartial(object.correctedMargin)
            : undefined;
        return message;
    },
};
function createBaseGetUserTariffRequest() {
    return {};
}
exports.GetUserTariffRequest = {
    encode(_, writer = new wire_1.BinaryWriter()) {
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof wire_1.BinaryReader ? input : new wire_1.BinaryReader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseGetUserTariffRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(_) {
        return {};
    },
    toJSON(_) {
        const obj = {};
        return obj;
    },
    create(base) {
        return exports.GetUserTariffRequest.fromPartial(base !== null && base !== void 0 ? base : {});
    },
    fromPartial(_) {
        const message = createBaseGetUserTariffRequest();
        return message;
    },
};
function createBaseGetUserTariffResponse() {
    return { unaryLimits: [], streamLimits: [] };
}
exports.GetUserTariffResponse = {
    encode(message, writer = new wire_1.BinaryWriter()) {
        for (const v of message.unaryLimits) {
            exports.UnaryLimit.encode(v, writer.uint32(10).fork()).join();
        }
        for (const v of message.streamLimits) {
            exports.StreamLimit.encode(v, writer.uint32(18).fork()).join();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof wire_1.BinaryReader ? input : new wire_1.BinaryReader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseGetUserTariffResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1: {
                    if (tag !== 10) {
                        break;
                    }
                    message.unaryLimits.push(exports.UnaryLimit.decode(reader, reader.uint32()));
                    continue;
                }
                case 2: {
                    if (tag !== 18) {
                        break;
                    }
                    message.streamLimits.push(exports.StreamLimit.decode(reader, reader.uint32()));
                    continue;
                }
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            unaryLimits: globalThis.Array.isArray(object === null || object === void 0 ? void 0 : object.unaryLimits)
                ? object.unaryLimits.map((e) => exports.UnaryLimit.fromJSON(e))
                : [],
            streamLimits: globalThis.Array.isArray(object === null || object === void 0 ? void 0 : object.streamLimits)
                ? object.streamLimits.map((e) => exports.StreamLimit.fromJSON(e))
                : [],
        };
    },
    toJSON(message) {
        var _a, _b;
        const obj = {};
        if ((_a = message.unaryLimits) === null || _a === void 0 ? void 0 : _a.length) {
            obj.unaryLimits = message.unaryLimits.map((e) => exports.UnaryLimit.toJSON(e));
        }
        if ((_b = message.streamLimits) === null || _b === void 0 ? void 0 : _b.length) {
            obj.streamLimits = message.streamLimits.map((e) => exports.StreamLimit.toJSON(e));
        }
        return obj;
    },
    create(base) {
        return exports.GetUserTariffResponse.fromPartial(base !== null && base !== void 0 ? base : {});
    },
    fromPartial(object) {
        var _a, _b;
        const message = createBaseGetUserTariffResponse();
        message.unaryLimits = ((_a = object.unaryLimits) === null || _a === void 0 ? void 0 : _a.map((e) => exports.UnaryLimit.fromPartial(e))) || [];
        message.streamLimits = ((_b = object.streamLimits) === null || _b === void 0 ? void 0 : _b.map((e) => exports.StreamLimit.fromPartial(e))) || [];
        return message;
    },
};
function createBaseUnaryLimit() {
    return { limitPerMinute: 0, methods: [] };
}
exports.UnaryLimit = {
    encode(message, writer = new wire_1.BinaryWriter()) {
        if (message.limitPerMinute !== 0) {
            writer.uint32(8).int32(message.limitPerMinute);
        }
        for (const v of message.methods) {
            writer.uint32(18).string(v);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof wire_1.BinaryReader ? input : new wire_1.BinaryReader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseUnaryLimit();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1: {
                    if (tag !== 8) {
                        break;
                    }
                    message.limitPerMinute = reader.int32();
                    continue;
                }
                case 2: {
                    if (tag !== 18) {
                        break;
                    }
                    message.methods.push(reader.string());
                    continue;
                }
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            limitPerMinute: isSet(object.limitPerMinute) ? globalThis.Number(object.limitPerMinute) : 0,
            methods: globalThis.Array.isArray(object === null || object === void 0 ? void 0 : object.methods) ? object.methods.map((e) => globalThis.String(e)) : [],
        };
    },
    toJSON(message) {
        var _a;
        const obj = {};
        if (message.limitPerMinute !== 0) {
            obj.limitPerMinute = Math.round(message.limitPerMinute);
        }
        if ((_a = message.methods) === null || _a === void 0 ? void 0 : _a.length) {
            obj.methods = message.methods;
        }
        return obj;
    },
    create(base) {
        return exports.UnaryLimit.fromPartial(base !== null && base !== void 0 ? base : {});
    },
    fromPartial(object) {
        var _a, _b;
        const message = createBaseUnaryLimit();
        message.limitPerMinute = (_a = object.limitPerMinute) !== null && _a !== void 0 ? _a : 0;
        message.methods = ((_b = object.methods) === null || _b === void 0 ? void 0 : _b.map((e) => e)) || [];
        return message;
    },
};
function createBaseStreamLimit() {
    return { limit: 0, streams: [], open: 0 };
}
exports.StreamLimit = {
    encode(message, writer = new wire_1.BinaryWriter()) {
        if (message.limit !== 0) {
            writer.uint32(8).int32(message.limit);
        }
        for (const v of message.streams) {
            writer.uint32(18).string(v);
        }
        if (message.open !== 0) {
            writer.uint32(24).int32(message.open);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof wire_1.BinaryReader ? input : new wire_1.BinaryReader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseStreamLimit();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1: {
                    if (tag !== 8) {
                        break;
                    }
                    message.limit = reader.int32();
                    continue;
                }
                case 2: {
                    if (tag !== 18) {
                        break;
                    }
                    message.streams.push(reader.string());
                    continue;
                }
                case 3: {
                    if (tag !== 24) {
                        break;
                    }
                    message.open = reader.int32();
                    continue;
                }
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            limit: isSet(object.limit) ? globalThis.Number(object.limit) : 0,
            streams: globalThis.Array.isArray(object === null || object === void 0 ? void 0 : object.streams) ? object.streams.map((e) => globalThis.String(e)) : [],
            open: isSet(object.open) ? globalThis.Number(object.open) : 0,
        };
    },
    toJSON(message) {
        var _a;
        const obj = {};
        if (message.limit !== 0) {
            obj.limit = Math.round(message.limit);
        }
        if ((_a = message.streams) === null || _a === void 0 ? void 0 : _a.length) {
            obj.streams = message.streams;
        }
        if (message.open !== 0) {
            obj.open = Math.round(message.open);
        }
        return obj;
    },
    create(base) {
        return exports.StreamLimit.fromPartial(base !== null && base !== void 0 ? base : {});
    },
    fromPartial(object) {
        var _a, _b, _c;
        const message = createBaseStreamLimit();
        message.limit = (_a = object.limit) !== null && _a !== void 0 ? _a : 0;
        message.streams = ((_b = object.streams) === null || _b === void 0 ? void 0 : _b.map((e) => e)) || [];
        message.open = (_c = object.open) !== null && _c !== void 0 ? _c : 0;
        return message;
    },
};
function createBaseGetInfoRequest() {
    return {};
}
exports.GetInfoRequest = {
    encode(_, writer = new wire_1.BinaryWriter()) {
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof wire_1.BinaryReader ? input : new wire_1.BinaryReader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseGetInfoRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(_) {
        return {};
    },
    toJSON(_) {
        const obj = {};
        return obj;
    },
    create(base) {
        return exports.GetInfoRequest.fromPartial(base !== null && base !== void 0 ? base : {});
    },
    fromPartial(_) {
        const message = createBaseGetInfoRequest();
        return message;
    },
};
function createBaseGetInfoResponse() {
    return { premStatus: false, qualStatus: false, qualifiedForWorkWith: [], tariff: "" };
}
exports.GetInfoResponse = {
    encode(message, writer = new wire_1.BinaryWriter()) {
        if (message.premStatus !== false) {
            writer.uint32(8).bool(message.premStatus);
        }
        if (message.qualStatus !== false) {
            writer.uint32(16).bool(message.qualStatus);
        }
        for (const v of message.qualifiedForWorkWith) {
            writer.uint32(26).string(v);
        }
        if (message.tariff !== "") {
            writer.uint32(34).string(message.tariff);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof wire_1.BinaryReader ? input : new wire_1.BinaryReader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseGetInfoResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1: {
                    if (tag !== 8) {
                        break;
                    }
                    message.premStatus = reader.bool();
                    continue;
                }
                case 2: {
                    if (tag !== 16) {
                        break;
                    }
                    message.qualStatus = reader.bool();
                    continue;
                }
                case 3: {
                    if (tag !== 26) {
                        break;
                    }
                    message.qualifiedForWorkWith.push(reader.string());
                    continue;
                }
                case 4: {
                    if (tag !== 34) {
                        break;
                    }
                    message.tariff = reader.string();
                    continue;
                }
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            premStatus: isSet(object.premStatus) ? globalThis.Boolean(object.premStatus) : false,
            qualStatus: isSet(object.qualStatus) ? globalThis.Boolean(object.qualStatus) : false,
            qualifiedForWorkWith: globalThis.Array.isArray(object === null || object === void 0 ? void 0 : object.qualifiedForWorkWith)
                ? object.qualifiedForWorkWith.map((e) => globalThis.String(e))
                : [],
            tariff: isSet(object.tariff) ? globalThis.String(object.tariff) : "",
        };
    },
    toJSON(message) {
        var _a;
        const obj = {};
        if (message.premStatus !== false) {
            obj.premStatus = message.premStatus;
        }
        if (message.qualStatus !== false) {
            obj.qualStatus = message.qualStatus;
        }
        if ((_a = message.qualifiedForWorkWith) === null || _a === void 0 ? void 0 : _a.length) {
            obj.qualifiedForWorkWith = message.qualifiedForWorkWith;
        }
        if (message.tariff !== "") {
            obj.tariff = message.tariff;
        }
        return obj;
    },
    create(base) {
        return exports.GetInfoResponse.fromPartial(base !== null && base !== void 0 ? base : {});
    },
    fromPartial(object) {
        var _a, _b, _c, _d;
        const message = createBaseGetInfoResponse();
        message.premStatus = (_a = object.premStatus) !== null && _a !== void 0 ? _a : false;
        message.qualStatus = (_b = object.qualStatus) !== null && _b !== void 0 ? _b : false;
        message.qualifiedForWorkWith = ((_c = object.qualifiedForWorkWith) === null || _c === void 0 ? void 0 : _c.map((e) => e)) || [];
        message.tariff = (_d = object.tariff) !== null && _d !== void 0 ? _d : "";
        return message;
    },
};
exports.UsersServiceDefinition = {
    name: "UsersService",
    fullName: "tinkoff.public.invest.api.contract.v1.UsersService",
    methods: {
        /** Получить счета пользователя. */
        getAccounts: {
            name: "GetAccounts",
            requestType: exports.GetAccountsRequest,
            requestStream: false,
            responseType: exports.GetAccountsResponse,
            responseStream: false,
            options: {},
        },
        /** Рассчитать маржинальные показатели по счёту. */
        getMarginAttributes: {
            name: "GetMarginAttributes",
            requestType: exports.GetMarginAttributesRequest,
            requestStream: false,
            responseType: exports.GetMarginAttributesResponse,
            responseStream: false,
            options: {},
        },
        /** Запросить тариф пользователя. */
        getUserTariff: {
            name: "GetUserTariff",
            requestType: exports.GetUserTariffRequest,
            requestStream: false,
            responseType: exports.GetUserTariffResponse,
            responseStream: false,
            options: {},
        },
        /** Получить информацию о пользователе. */
        getInfo: {
            name: "GetInfo",
            requestType: exports.GetInfoRequest,
            requestStream: false,
            responseType: exports.GetInfoResponse,
            responseStream: false,
            options: {},
        },
    },
};
function toTimestamp(date) {
    const seconds = Math.trunc(date.getTime() / 1000);
    const nanos = (date.getTime() % 1000) * 1000000;
    return { seconds, nanos };
}
function fromTimestamp(t) {
    let millis = (t.seconds || 0) * 1000;
    millis += (t.nanos || 0) / 1000000;
    return new globalThis.Date(millis);
}
function fromJsonTimestamp(o) {
    if (o instanceof globalThis.Date) {
        return o;
    }
    else if (typeof o === "string") {
        return new globalThis.Date(o);
    }
    else {
        return fromTimestamp(timestamp_1.Timestamp.fromJSON(o));
    }
}
function isSet(value) {
    return value !== null && value !== undefined;
}
