// @tinkoff/invest-js/build/generated/tinkoff/public/invest/api/contract/v1/GetTechAnalysisRequest.js
"use strict";
// Original file: investAPI/src/docs/contracts/marketdata.proto
Object.defineProperty(exports, "__esModule", { value: true });
exports._tinkoff_public_invest_api_contract_v1_GetTechAnalysisRequest_TypeOfPrice = exports._tinkoff_public_invest_api_contract_v1_GetTechAnalysisRequest_IndicatorType = exports._tinkoff_public_invest_api_contract_v1_GetTechAnalysisRequest_IndicatorInterval = void 0;
// Original file: investAPI/src/docs/contracts/marketdata.proto
exports._tinkoff_public_invest_api_contract_v1_GetTechAnalysisRequest_IndicatorInterval = {
    INDICATOR_INTERVAL_UNSPECIFIED: 'INDICATOR_INTERVAL_UNSPECIFIED',
    INDICATOR_INTERVAL_ONE_MINUTE: 'INDICATOR_INTERVAL_ONE_MINUTE',
    INDICATOR_INTERVAL_FIVE_MINUTES: 'INDICATOR_INTERVAL_FIVE_MINUTES',
    INDICATOR_INTERVAL_FIFTEEN_MINUTES: 'INDICATOR_INTERVAL_FIFTEEN_MINUTES',
    INDICATOR_INTERVAL_ONE_HOUR: 'INDICATOR_INTERVAL_ONE_HOUR',
    INDICATOR_INTERVAL_ONE_DAY: 'INDICATOR_INTERVAL_ONE_DAY',
    INDICATOR_INTERVAL_2_MIN: 'INDICATOR_INTERVAL_2_MIN',
    INDICATOR_INTERVAL_3_MIN: 'INDICATOR_INTERVAL_3_MIN',
    INDICATOR_INTERVAL_10_MIN: 'INDICATOR_INTERVAL_10_MIN',
    INDICATOR_INTERVAL_30_MIN: 'INDICATOR_INTERVAL_30_MIN',
    INDICATOR_INTERVAL_2_HOUR: 'INDICATOR_INTERVAL_2_HOUR',
    INDICATOR_INTERVAL_4_HOUR: 'INDICATOR_INTERVAL_4_HOUR',
    INDICATOR_INTERVAL_WEEK: 'INDICATOR_INTERVAL_WEEK',
    INDICATOR_INTERVAL_MONTH: 'INDICATOR_INTERVAL_MONTH',
};
// Original file: investAPI/src/docs/contracts/marketdata.proto
exports._tinkoff_public_invest_api_contract_v1_GetTechAnalysisRequest_IndicatorType = {
    INDICATOR_TYPE_UNSPECIFIED: 'INDICATOR_TYPE_UNSPECIFIED',
    INDICATOR_TYPE_BB: 'INDICATOR_TYPE_BB',
    INDICATOR_TYPE_EMA: 'INDICATOR_TYPE_EMA',
    INDICATOR_TYPE_RSI: 'INDICATOR_TYPE_RSI',
    INDICATOR_TYPE_MACD: 'INDICATOR_TYPE_MACD',
    INDICATOR_TYPE_SMA: 'INDICATOR_TYPE_SMA',
};
// Original file: investAPI/src/docs/contracts/marketdata.proto
exports._tinkoff_public_invest_api_contract_v1_GetTechAnalysisRequest_TypeOfPrice = {
    TYPE_OF_PRICE_UNSPECIFIED: 'TYPE_OF_PRICE_UNSPECIFIED',
    TYPE_OF_PRICE_CLOSE: 'TYPE_OF_PRICE_CLOSE',
    TYPE_OF_PRICE_OPEN: 'TYPE_OF_PRICE_OPEN',
    TYPE_OF_PRICE_HIGH: 'TYPE_OF_PRICE_HIGH',
    TYPE_OF_PRICE_LOW: 'TYPE_OF_PRICE_LOW',
    TYPE_OF_PRICE_AVG: 'TYPE_OF_PRICE_AVG',
};
