// @tinkoff/invest-js/build/generated/tinkoff/public/invest/api/contract/v1/OrderStateStreamResponse.js
"use strict";
// Original file: investAPI/src/docs/contracts/orders.proto
Object.defineProperty(exports, "__esModule", { value: true });
exports._tinkoff_public_invest_api_contract_v1_OrderStateStreamResponse_StatusCauseInfo = exports._tinkoff_public_invest_api_contract_v1_OrderStateStreamResponse_MarkerType = void 0;
// Original file: investAPI/src/docs/contracts/orders.proto
exports._tinkoff_public_invest_api_contract_v1_OrderStateStreamResponse_MarkerType = {
    MARKER_UNKNOWN: 'MARKER_UNKNOWN',
    MARKER_BROKER: 'MARKER_BROKER',
    MARKER_CHAT: 'MARKER_CHAT',
    MARKER_PAPER: 'MARKER_PAPER',
    MARKER_MARGIN: 'MARKER_MARGIN',
    MARKER_TKBNM: 'MARKER_TKBNM',
    MARKER_SHORT: 'MARKER_SHORT',
    MARKER_SPECMM: 'MARKER_SPECMM',
    MARKER_PO: 'MARKER_PO',
};
// Original file: investAPI/src/docs/contracts/orders.proto
exports._tinkoff_public_invest_api_contract_v1_OrderStateStreamResponse_StatusCauseInfo = {
    CAUSE_UNSPECIFIED: 'CAUSE_UNSPECIFIED',
    CAUSE_CANCELLED_BY_CLIENT: 'CAUSE_CANCELLED_BY_CLIENT',
    CAUSE_CANCELLED_BY_EXCHANGE: 'CAUSE_CANCELLED_BY_EXCHANGE',
    CAUSE_CANCELLED_NOT_ENOUGH_POSITION: 'CAUSE_CANCELLED_NOT_ENOUGH_POSITION',
    CAUSE_CANCELLED_BY_CLIENT_BLOCK: 'CAUSE_CANCELLED_BY_CLIENT_BLOCK',
    CAUSE_REJECTED_BY_BROKER: 'CAUSE_REJECTED_BY_BROKER',
    CAUSE_REJECTED_BY_EXCHANGE: 'CAUSE_REJECTED_BY_EXCHANGE',
    CAUSE_CANCELLED_BY_BROKER: 'CAUSE_CANCELLED_BY_BROKER',
};
