// @tinkoff/invest-js/build/generated/tinkoff/public/invest/api/contract/v1/SecurityTradingStatus.js
"use strict";
// Original file: investAPI/src/docs/contracts/common.proto
Object.defineProperty(exports, "__esModule", { value: true });
exports.SecurityTradingStatus = void 0;
exports.SecurityTradingStatus = {
    SECURITY_TRADING_STATUS_UNSPECIFIED: 'SECURITY_TRADING_STATUS_UNSPECIFIED',
    SECURITY_TRADING_STATUS_NOT_AVAILABLE_FOR_TRADING: 'SECURITY_TRADING_STATUS_NOT_AVAILABLE_FOR_TRADING',
    SECURITY_TRADING_STATUS_OPENING_PERIOD: 'SECURITY_TRADING_STATUS_OPENING_PERIOD',
    SECURITY_TRADING_STATUS_CLOSING_PERIOD: 'SECURITY_TRADING_STATUS_CLOSING_PERIOD',
    SECURITY_TRADING_STATUS_BREAK_IN_TRADING: 'SECURITY_TRADING_STATUS_BREAK_IN_TRADING',
    SECURITY_TRADING_STATUS_NORMAL_TRADING: 'SECURITY_TRADING_STATUS_NORMAL_TRADING',
    SECURITY_TRADING_STATUS_CLOSING_AUCTION: 'SECURITY_TRADING_STATUS_CLOSING_AUCTION',
    SECURITY_TRADING_STATUS_DARK_POOL_AUCTION: 'SECURITY_TRADING_STATUS_DARK_POOL_AUCTION',
    SECURITY_TRADING_STATUS_DISCRETE_AUCTION: 'SECURITY_TRADING_STATUS_DISCRETE_AUCTION',
    SECURITY_TRADING_STATUS_OPENING_AUCTION_PERIOD: 'SECURITY_TRADING_STATUS_OPENING_AUCTION_PERIOD',
    SECURITY_TRADING_STATUS_TRADING_AT_CLOSING_AUCTION_PRICE: 'SECURITY_TRADING_STATUS_TRADING_AT_CLOSING_AUCTION_PRICE',
    SECURITY_TRADING_STATUS_SESSION_ASSIGNED: 'SECURITY_TRADING_STATUS_SESSION_ASSIGNED',
    SECURITY_TRADING_STATUS_SESSION_CLOSE: 'SECURITY_TRADING_STATUS_SESSION_CLOSE',
    SECURITY_TRADING_STATUS_SESSION_OPEN: 'SECURITY_TRADING_STATUS_SESSION_OPEN',
    SECURITY_TRADING_STATUS_DEALER_NORMAL_TRADING: 'SECURITY_TRADING_STATUS_DEALER_NORMAL_TRADING',
    SECURITY_TRADING_STATUS_DEALER_BREAK_IN_TRADING: 'SECURITY_TRADING_STATUS_DEALER_BREAK_IN_TRADING',
    SECURITY_TRADING_STATUS_DEALER_NOT_AVAILABLE_FOR_TRADING: 'SECURITY_TRADING_STATUS_DEALER_NOT_AVAILABLE_FOR_TRADING',
};
