// @tinkoff/invest-js/build/generated/tinkoff/public/invest/api/contract/v1/SubscriptionStatus.js
"use strict";
// Original file: investAPI/src/docs/contracts/marketdata.proto
Object.defineProperty(exports, "__esModule", { value: true });
exports.SubscriptionStatus = void 0;
exports.SubscriptionStatus = {
    SUBSCRIPTION_STATUS_UNSPECIFIED: 'SUBSCRIPTION_STATUS_UNSPECIFIED',
    SUBSCRIPTION_STATUS_SUCCESS: 'SUBSCRIPTION_STATUS_SUCCESS',
    SUBSCRIPTION_STATUS_INSTRUMENT_NOT_FOUND: 'SUBSCRIPTION_STATUS_INSTRUMENT_NOT_FOUND',
    SUBSCRIPTION_STATUS_SUBSCRIPTION_ACTION_IS_INVALID: 'SUBSCRIPTION_STATUS_SUBSCRIPTION_ACTION_IS_INVALID',
    SUBSCRIPTION_STATUS_DEPTH_IS_INVALID: 'SUBSCRIPTION_STATUS_DEPTH_IS_INVALID',
    SUBSCRIPTION_STATUS_INTERVAL_IS_INVALID: 'SUBSCRIPTION_STATUS_INTERVAL_IS_INVALID',
    SUBSCRIPTION_STATUS_LIMIT_IS_EXCEEDED: 'SUBSCRIPTION_STATUS_LIMIT_IS_EXCEEDED',
    SUBSCRIPTION_STATUS_INTERNAL_ERROR: 'SUBSCRIPTION_STATUS_INTERNAL_ERROR',
    SUBSCRIPTION_STATUS_TOO_MANY_REQUESTS: 'SUBSCRIPTION_STATUS_TOO_MANY_REQUESTS',
    SUBSCRIPTION_STATUS_SUBSCRIPTION_NOT_FOUND: 'SUBSCRIPTION_STATUS_SUBSCRIPTION_NOT_FOUND',
    SUBSCRIPTION_STATUS_SOURCE_IS_INVALID: 'SUBSCRIPTION_STATUS_SOURCE_IS_INVALID',
};
