// @tinkoff/invest-js/build/generated/tinkoff/public/invest/api/contract/v1/SubscriptionInterval.js
"use strict";
// Original file: investAPI/src/docs/contracts/marketdata.proto
Object.defineProperty(exports, "__esModule", { value: true });
exports.SubscriptionInterval = void 0;
exports.SubscriptionInterval = {
    SUBSCRIPTION_INTERVAL_UNSPECIFIED: 'SUBSCRIPTION_INTERVAL_UNSPECIFIED',
    SUBSCRIPTION_INTERVAL_ONE_MINUTE: 'SUBSCRIPTION_INTERVAL_ONE_MINUTE',
    SUBSCRIPTION_INTERVAL_FIVE_MINUTES: 'SUBSCRIPTION_INTERVAL_FIVE_MINUTES',
    SUBSCRIPTION_INTERVAL_FIFTEEN_MINUTES: 'SUBSCRIPTION_INTERVAL_FIFTEEN_MINUTES',
    SUBSCRIPTION_INTERVAL_ONE_HOUR: 'SUBSCRIPTION_INTERVAL_ONE_HOUR',
    SUBSCRIPTION_INTERVAL_ONE_DAY: 'SUBSCRIPTION_INTERVAL_ONE_DAY',
    SUBSCRIPTION_INTERVAL_2_MIN: 'SUBSCRIPTION_INTERVAL_2_MIN',
    SUBSCRIPTION_INTERVAL_3_MIN: 'SUBSCRIPTION_INTERVAL_3_MIN',
    SUBSCRIPTION_INTERVAL_10_MIN: 'SUBSCRIPTION_INTERVAL_10_MIN',
    SUBSCRIPTION_INTERVAL_30_MIN: 'SUBSCRIPTION_INTERVAL_30_MIN',
    SUBSCRIPTION_INTERVAL_2_HOUR: 'SUBSCRIPTION_INTERVAL_2_HOUR',
    SUBSCRIPTION_INTERVAL_4_HOUR: 'SUBSCRIPTION_INTERVAL_4_HOUR',
    SUBSCRIPTION_INTERVAL_WEEK: 'SUBSCRIPTION_INTERVAL_WEEK',
    SUBSCRIPTION_INTERVAL_MONTH: 'SUBSCRIPTION_INTERVAL_MONTH',
};
